### PR TITLE
refactor(FOROME-963): created the new async store for ws_list method

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-963
PR into https://github.com/ForomePlatform/Anfisa-React-Client/pull/608

In addition:
1. Created the new main-table store
2. Removed unused types from index.d.ts

### CODE REVIEW

File: src/store/ws/main-table.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    reaction(
+      () => this.wsRecords,
+      () => {
+        this.indexFilteredNo = 0
+        this.fetchFilteredTabReportAsync()
+      },
+    )
+  }
+
+  private get wsListQuery(): IWsListQuery {
```
Comment (AlMaQntr) [inline src/store/ws/main-table.ts]: I think you can use `filterPresetStore.activePreset` and use it as `filter` param for query.
But you should avoid to pass `conditions` when `activePreset` is not empty

**Review by AlMaQntr — APPROVED** [review]

File: src/store/ws/main-table.ts (hunk 1 above)
Comment (Lapk1n) [reply src/store/ws/main-table.ts]: Fixed

**Review by QYaroslavTrefilov — APPROVED** [review]

File: src/store/ws/main-table.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
+import { makeAutoObservable, reaction, runInAction, toJS } from 'mobx'
```
Comment (AndreyUstyumenko) [inline src/store/ws/main-table.ts]: main-table.store.ts

File: src/store/ws/main-table.ts [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  public get filteredNo(): number[] {
+    return this.wsRecords
+      ? this.wsRecords.map((variant: { no: number }) => variant.no)
+      : []
+  }
+
+  public get statAmount(): TItemsCount | undefined {
+    return filterStore.conditions.length || zoneStore.zone.length
+      ? this.wsList.data?.['filtered-counts']
+      : this.wsList.data?.['total-counts']
```
Comment (AndreyUstyumenko) [inline src/store/ws/main-table.ts]: I would suggest to create an adapter for wsDatasetSupportProvider.getWsList to convert kebab-cased props to camel

File: src/store/dataset.ts [hunk 4]
```diff
@@ -1,255 +1,61 @@
...
-
-      return
-    }
-
-    const tabReport = await datasetProvider.getTabReport({
-      ds: dsName,
-      schema: 'xbr',
-      seq,
+  async fetchDsinfoAsync(name: string) {
+    const response = await fetch(getApiUrl(`dsinfo?ds=${name}`), {
```
Comment (AndreyUstyumenko) [inline src/store/dataset.ts]: DatasetProvider.getDsInfo

File: src/store/ws/main-table.ts (hunk 2 above)
Comment (Lapk1n) [reply src/store/ws/main-table.ts]: fixed

File: src/store/dataset.ts (hunk 4 above)
Comment (Lapk1n) [reply src/store/dataset.ts]: fixed

File: src/store/ws/main-table.ts (hunk 3 above)
Comment (Lapk1n) [reply src/store/ws/main-table.ts]: added

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**